apps/qubit/modules/user/actions/clipboardExportAction.class.php

```diff
@@ -210,16 +210,23 @@ public function execute($request)
       else
       {
         $job = QubitJob::runJob($jobName, $options);
+        
+        $responseData = array();
 
-        // TODO: change this to not use user session
+        // Generate, store and return a token to associate unauthenticated users with
+        // their export jobs to be able to download the result later and delete the job.
         if (!$this->context->user->isAuthenticated())
         {
+          $property = $job->generateUserTokenProperty();
+          $responseData['token'] = $property->value;
+
+          // TODO: REMOVE WHEN DONE!!
           $manager = new QubitUnauthenticatedUserJobManager($this->context->user);
           $manager->addJobAssociation($job);
         }
 
-        $message = '<p>';
-        $message .= $this->context->i18n->__(
+        $responseData['success'] = '<p>';
+        $responseData['success'] .= $this->context->i18n->__(
           '%1%Your %2% export package is being built.%3% ' . ($this->context->user->isAuthenticated() ? 'The %4%job management page%5% will show progress and a download link when complete.' : 'Please %4%refresh the page%5% to see progress and a download link when complete.'), 
```

Comment:
I also don't know how the `__()` function will render the ternary condition here, but I don't think it will allow translation of the conditional strings.

Reply:
Yes, this is all quite a mess, I didn't even want to look at it during the main work but I'll tweak it now. Thanks!

Reply:
Still not perfect but it should be easy to read now and give some context to translators.